### PR TITLE
Adding required and validate attributes and functionalities

### PR DESCRIPTION
**Pull request by dotpointer**
This pull request adds required and validate capabilities to the component.

Tested and seems to work, but please review and come with suggestions for improvement.

2 approves or a merge requested.

This functionality is needed by Redmine issue #18951.

### CODE REVIEW

**Comment by dotpointer** [comment]
Found a bug with validation not passing, please repull if you have cloned already.

**Comment by dotpointer** [comment]
The involvement of paper-autocomplete was because it already had the error displayment - the red border - built in. But it is not an optimal solution and I found another bug, it will (of course) invalidly pass validation if something is filled in the paper-autocomplete, but nothing is selected.

The idea was also to disable required state on the paper-autocomplete if something was selected, therefore would required + validate not fail the input, because it then is no longer required.

validate always returning false was a bug that I corrected just before you posted the comment.

required=true + selectedItems < 0 seemed usable to trigger the error state, even though it should not work with the paper-autocomplete as it does at the moment?

**Comment by nomego** [comment]
And set a custom `errorMessage`

**Comment by dotpointer** [comment]
I have now pushed a change that removes required for the paper-autocomplete and instead sets `invalid` and `errorMessage` on it. Error is displayed if required is set and the amount of selected items are none. I have also added to properties I mentioned in #24, the `minSelected` and `maxSelected` to allow control of the minimum and maximum quantity of selected items. The added error messages required addition of the translation behavior.

Not sure that I followed the usage of `allowedPattern`, was that instead of `invalid` as a workaround (like the usage of `pattern` mentioned here: this https://github.com/ellipticaljs/paper-autocomplete/issues/50) to trigger invalid state on the paper-autocomplete?

File: paper-autocomplete-chips.html [hunk 1]
```diff
@@ -135,6 +136,22 @@
 					value: 'Search'
 				},
 
+				/**
+				 * `<paper-autocomplete>` `invalid`
+				 */
+				 _acInvalid: {
```
Comment (nomego) [inline paper-autocomplete-chips.html]: Not used unless you drop the `_`

File: paper-autocomplete-chips.html [hunk 2]
```diff
@@ -143,6 +160,30 @@
 					value: 0
 				},
 
+				/**
+				 * Minimum limit of selected items, set to 0 to disable.
+				 */
+				 minSelected: {
```
Comment (nomego) [inline paper-autocomplete-chips.html]: `<input>` attributes are `min` / `max` - we might want to use the same?

File: paper-autocomplete-chips.html [hunk 3]
```diff
@@ -332,7 +375,30 @@
 						}
 					});
 				});
-			}
+			},
+
+			validate() {
+				if (this.required && !this.selectedItems.length) {
```
Comment (nomego) [inline paper-autocomplete-chips.html]: `this.selectedItems.length === 0` ?

File: paper-autocomplete-chips.html [hunk 4]
```diff
@@ -332,7 +375,30 @@
...
+
+				if (this.maxSelected && this.maxSelected < this.selectedItems.length) {
+					this.set('acInvalid', true);
+					this.set('errorMessage', this.gettext('Select maximum {0} in the list.', this.maxSelected));
+					return false;
+				}
+
+				this.set('acInvalid', false);
+				this.set('errorMessage', '');
+				return true;			}
```
Comment (nomego) [inline paper-autocomplete-chips.html]: `}` new line

File: paper-autocomplete-chips.html (hunk 2 above)
Comment (dotpointer) [reply paper-autocomplete-chips.html]: Renamed them to min and max.

File: paper-autocomplete-chips.html (hunk 3 above)
Comment (dotpointer) [reply paper-autocomplete-chips.html]: Yes, used this instead.

File: paper-autocomplete-chips.html (hunk 4 above)
Comment (dotpointer) [reply paper-autocomplete-chips.html]: Added new line.

**Comment by nomego** [comment]
Looks good! But tests are failing since cosmoz-i18next isn't installed for this component.
So fix bower.json and we can merge.

**Comment by dotpointer** [comment]
@nomego Added cosmoz-i18next to bower.json in an amend. All tests but codeclimate/total-coverage which concerns a FIXME statement seems to pass now on Travis.

**Pull request merged**